File: C/Program15.c
/////////////////////////////////////////////////////////////////////////
//                                                                     //
//  File name :     Program15.c                                        //
//  Description :   Used to Display Factors of given number            //
//  Author :        Pallavi Omprakash Malewar                          //
//  Date :          19/07/2025                                         //
//                                                                     //
/////////////////////////////////////////////////////////////////////////


#include<stdio.h>

void DisplayFactors(int iNo)
{
    int iCount = 0;

    if(iNo < 0)
    {
        iNo = -iNo;     // Updator 
    }
    for(iCount = 1 ; iCount <= (iNo / 2) ; iCount++)
    {
        if((iNo % iCount) == 0)
        {
            printf("%d\n" , iCount);
        }
    }
    
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number to display factors : ");
    scanf("%d" , &iValue);

    DisplayFactors(iValue);
    
    return 0;
}

// Time Complexity O(N/2)

File: JAVA/Program15.java
//////////////////////////////////////////////////////////////////////////////////////////
//                                                                                      //
//  File name :     Program15.java                                                      //
//  Description :   Used to Display Even Number from given Number                       //
//  Author :        Pallavi Omprakash Malewar                                           //
//  Date :          25/08/2025                                                          //
//                                                                                      //
//////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Digits
{
    public void displayEven(int iNo)
    {
        int iDigit = 0, iCount = 0;
        boolean hasEven = false;

        if(iNo < 0)
        {
            iNo = -iNo;     // Filter
        }

        if(iNo == 0)
        {
            System.out.println("0");
            hasEven = true;
            return;
        }

        while(iNo != 0)
        {
            iDigit = iNo % 10;

            if((iDigit % 2) == 0)
            {
                iCount++;
                System.out.print(iDigit + "\t");
                hasEven = true;
            }

            iNo = iNo / 10;
        }

        if(!hasEven)
        {
            System.out.pritnln("Even Digits are Not Found");
        }

    }
}

class Program15
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.print("Enter Number : ");

        if(!sobj.hasNextInt())
        {
            System.out.println("Invalid Input.");
            return;

        }

        iValue = sobj.nextInt();

        Digits dobj = new Digits();

        dobj.displayEven(iValue);
    }
}